src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -715,21 +738,32 @@ private static bool IsPathRootedSafe(string path)
             return path.IndexOfAny(Path.GetInvalidPathChars()) == -1 && Path.IsPathRooted(path);
         }
 
-        private void SetHttpWarningVisibilityForSelectSource(PackageSourceContextInfo selectedSource)
+        private void SetHttpErrorVisibilityForSelectSource(PackageSourceContextInfo selectedSource)
         {
             var source = new PackageSource(selectedSource.Source, selectedSource.Name);
             source.AllowInsecureConnections = selectedSource.AllowInsecureConnections;
 
-            // Warn if the selected source is http, and the AllowInsecureConnections for this source is set to false. 
+            HttpErrorOrWarning.Visible = true;
+            HttpErrorOrWarningIcon.Visible = true;
+            configurationLink.Visible = true;
```

Comment:
By setting these to true before checking if the source is http, not http, and insecure connections are not allowed, does the error flash on the screen before reaching the else case where they're set to false? I prefer to err on the side of only showing error messages when we've confirmed the state is an error state. That way, on slow or busy machines, you wouldn't see the errors show visible then disappear a brief moment later.

My suggestion then would be to not modify the current visibility state until after we've confirmed there's an error or not to prevent a potential blinking effect.